koku/masu/test/processor/parquet/test_parquet_report_processor.py

```diff
@@ -593,3 +595,166 @@ def test_create_daily_parquet(self, mock_write, mock_create_table):
         mock_create_table.assert_called()
+
+    @patch.object(ParquetReportProcessor, "parquet_ocp_on_cloud_path_s3", return_value="")
+    @patch.object(ParquetReportProcessor, "parquet_daily_path_s3", return_value="")
+    @patch.object(ParquetReportProcessor, "parquet_path_s3", return_value="")
+    @patch("masu.processor.parquet.parquet_report_processor.filter_s3_objects_less_than")
+    @patch.object(ParquetReportProcessor, "parquet_file_getter")
+    def test_prepare_parquet_s3_ocp_files_reports_already_processed(self, mock_s3_getter, mock_s3_filter, *_):
+        """Test raising ReportsAlreadyProcessed."""
+        mock_s3_getter.return_value = ["file1"]
+        mock_s3_filter.return_value = []
+
+        filename = Path("pod_usage.count.csv")
+        CostUsageReportManifest.objects.filter(id=self.ocp_manifest_id).update(operator_daily_reports=True)
+
+        report_processor = ParquetReportProcessor(
+            schema_name=self.schema,
+            report_path=filename,
+            provider_uuid=self.ocp_provider_uuid,
+            provider_type=Provider.PROVIDER_OCP,
+            manifest_id=self.ocp_manifest_id,
+            context={
+                "tracing_id": self.tracing_id,
+                "start_date": DateHelper().today,
+                "ocp_files_to_process": {
+                    filename.stem: {
+                        "meta_reportdatestart": "2023-01-01",
+                        "meta_reportnumhours": "2",
+                    }
+                },
+            },
+        )
+        with self.assertRaises(ReportsAlreadyProcessed):
+            report_processor.prepare_parquet_s3(filename)
+
+    @patch("masu.processor.parquet.parquet_report_processor.delete_s3_objects")
+    @patch.object(ParquetReportProcessor, "parquet_ocp_on_cloud_path_s3", return_value="")
+    @patch.object(ParquetReportProcessor, "parquet_daily_path_s3", return_value="")
+    @patch.object(ParquetReportProcessor, "parquet_path_s3", return_value="")
+    @patch("masu.processor.parquet.parquet_report_processor.filter_s3_objects_less_than")
+    @patch.object(ParquetReportProcessor, "parquet_file_getter")
+    def test_prepare_parquet_s3_ocp_files_reports_to_delete(self, mock_s3_getter, mock_s3_filter, *_):
+        """Test that s3-parquet-tracker is updated when a we delete s3 files."""
+        mock_s3_getter.return_value = ["file1"]
+        mock_s3_filter.return_value = ["file1"]
+
+        filename = Path("pod_usage.count.csv")
+        CostUsageReportManifest.objects.filter(id=self.ocp_manifest_id).update(operator_daily_reports=True)
+
+        report_processor = ParquetReportProcessor(
+            schema_name=self.schema,
+            report_path=filename,
+            provider_uuid=self.ocp_provider_uuid,
+            provider_type=Provider.PROVIDER_OCP,
+            manifest_id=self.ocp_manifest_id,
+            context={
+                "tracing_id": self.tracing_id,
+                "start_date": DateHelper().today,
```

Comment:
Using the current date in tests makes them non-deterministic. It's better to use deterministic values in tests.

Not a blocker since we do this a lot currently. It would be good to get away from this practice, though.

Reply:
The date is not used for the test, but it is required to instantiate the `ParquetReportProcessor`.